dask/dataframe/core.py

```diff
@@ -1741,6 +1741,30 @@ def __repr__(self):
                                           name=key_split(self._name),
                                           task=len(self.dask))
 
+    @derived_from(pd.Series)
+    def rename(self, index=None, inplace=False):
+        from pandas.api.types import is_scalar, is_list_like, is_dict_like
+        if is_scalar(index) or (is_list_like(index) and not is_dict_like(index)):
+            res = self if inplace else self.copy()
+            res.name = index
+        else:
+            res = self.map_partitions(M.rename, index)
+            if self.known_divisions:
```

Comment:
Hmm, this block seems to assume that the `index` mapping is monotonic.


```python
In [13]: s = pd.Series(['a', 'b', 'c'])

In [14]: dd.from_pandas(s, 2).rename(lambda x: -x).divisions
Out[14]: (0, -2)
```

which I think is problematic.

Allowing the callable / dict-like `index` may be difficult to do in general, which I'm guessing is why it wasn't previously implemented. I'm not sure how best to proceed here, since this would be a useful enhancement for (the common?) case of monotonic transformations.

Reply:
Good catch. A few options:

- Always drop the known divisions and delete this block.
- Check for monotonic output divisions, and drop if not sorted. This is a bit unsafe, because transformations that are sorted only for the divisions would assume sorted everywhere.
- Add a flag (default False), where True the divisions are inferred and checked to be sorted. Not sure what to call this.

I'd go with 1 or 3.

Reply:
I think 3 is best? For keyword, how about `sorted` like from `DataFrame.set_index`

```
sorted: bool, optional
    If the index column is already sorted in increasing order.
    Defaults to False
```

Is it too strange to have `sorted=True` mean that the *transformed* values are sorted?

Reply:
How about `sorted_index` instead?

Reply:
Done